modules/tensor_mechanics/src/actions/LineElementAction.C

```diff
@@ -0,0 +1,834 @@
+                             "for a static analysis with beam or truss elements. Also sets up aux "
+                             "variables, aux kernels, and consistent or nodal inertia kernels for "
+                             "dynamic analysis with beam elements.");
+
+  params.addParam<bool>(
+      "truss",
+      false,
+      "Set to true if the line elements are truss elemrnts instead of the default beam elements.");
+  params.addParam<bool>("add_variables",
+                        false,
+                        "Add the displacement variables for truss elements "
+                        "and both diplacement and rotation variables for "
+                        "beam elements.");
+  params.addParam<std::vector<VariableName>>(
+      "displacements", "The nonlinear displacement variables for the problem");
+
+  // Common material parameters between beam and truss
+  params.addCoupledVar(
+      "youngs_modulus",
+      "Young's modulus of the material. Can be supplied as either a number or a variable name.");
+
+  // Common geometry parameters between beam and truss
+  params.addCoupledVar(
+      "area",
+      "Cross-section area of the beam. Can be supplied as either a number or a variable name.");
+
+  // Truss specific parameters
+  params.addCoupledVar("temperature", 273, "Temperature in Kelvin");
+  params.addParam<Real>("thermal_expansion_coeff", 0.0, "Thermal expansion coefficient in 1/K");
+  params.addParam<Real>("temperature_ref", 273, "Reference temperature for thermal expansion in K");
+
+  // Beam Parameters
+  params += LineElementAction::beamParameters();
+
+  params.addParam<bool>(
+      "use_displaced_mesh", false, "Whether to use displaced mesh in the kernels");
+  // parameters specified here only appear in the input file sub-blocks of the
+  // Master action, not in the common parameters area
+  params.addParam<std::vector<SubdomainName>>(
+      "block",
+      "The list of ids of the blocks (subdomain) "
+      "that the stress divergence, inertia kernels and materials will be "
+      "applied to");
+  // Advanced
+  params.addParam<std::vector<AuxVariableName>>("save_in",
+                                                "The displacement and rotational residuals");
+  params.addParam<std::vector<AuxVariableName>>(
+      "diag_save_in", "The displacement and rotational diagonal preconditioner terms");
+  params.addParamNamesToGroup("block", "Advanced");
+  return params;
+}
+
+InputParameters
+LineElementAction::beamParameters()
+{
+  InputParameters params = emptyInputParameters();
+
+  params.addParam<std::vector<VariableName>>(
+      "rotations", "The rotations appropriate for the simulation geometry and coordinate system");
+
+  MooseEnum strainType("SMALL_STRAIN_AND_ROTATION FINITE_STRAIN_AND_SMALL_ROTATION "
+                       "SMALL_STRAIN_AND_FINITE_ROTATION FINITE_STRAIN_AND_ROTATION",
+                       "SMALL_STRAIN_AND_ROTATION");
```

Comment:
I'd propose these names:
```
SMALL_STRAIN_SMALL_ROTATION
FINITE_STRAIN_SMALL_ROTATION
SMALL_STRAIN_FINITE_ROTATION
FINITE_STRAIN_FINITE_ROTATION
SMALL_STRAIN_SMALL_ROTATION
```
Alternatively, could you have two separate parameters `strain` and `rotation` that could have values of `SMALL` and `FINITE`?